File: Homework1/Animal.java
import java.io.*;
import java.util.Random;

public class Animal
{
  public String name = "";
  public boolean isAwake = false;
  public boolean isFed = false;

	public void sleep()
	{
		System.out.println("Sleeping");
	}
	public int random_number_generator()
	{
		Random rand = new Random();
		int num = rand.nextInt(2);
		return num;
	}
  public void makeNoise(){
    System.out.println("");
  }
}

class Feline extends Animal
{
	public Feline(){}
	public void roam()
	{
		System.out.println("Wandering");
	}
}

class Cat extends Feline
{
	public Cat(String name){
    this.name=name;
  }
	public void random_action_generator()
	{
		int num = random_number_generator();
		if(num == 0)
		{
			roam();
		}
		else
		{
			System.out.println("Meow!");
		}
	}
	public void makeNoise()
	{
		random_action_generator();
	}
}

class Lion extends Feline
{
	public Lion(String name){
    this.name = name;
  }
	public void makeNoise()
	{
		System.out.println("Roar");
	}
}

class Tiger extends Feline
{
	public Tiger(String name){
    this.name = name;
  }
	public void makeNoise()
	{
		System.out.println("Grrrrrr");
	}
}

class Canine extends Animal
{
//	public Canine(){}
	public void roam()
	{
		System.out.println("Scampering");
	}
}

class Wolf extends Canine
{
	public Wolf(String name){
    this.name = name;
  }
	public void makeNoise()
	{
		System.out.println("Grrrrr");
	}
}

class Dog extends Canine
{
	public Dog(String name){
    this.name = name;
  }
	public void makeNoise()
	{
		System.out.println("Bark Bark");
	}
}

class Pachyderm extends Animal
{
	public Pachyderm(){}
	public void roam()
	{
		System.out.println("Stompping");
	}
}

class Hippo extends Pachyderm
{
	public Hippo(String name ){
    this.name = name;
  }
	public void makeNoise()
	{
		System.out.println("Whaaaa");
	}
}

class Rhino extends Pachyderm
{
	public Rhino(String name){
    this.name = name;
  }
	public void makeNoise()
	{
		System.out.println("Grumble Grumble");
	}
}

class Elephant extends Pachyderm
{
	public Elephant(String name){
    this.name = name;
  }
	public void makeNoise()
	{
		System.out.println("Raaaaaaa");
	}
}

File: Homework2/Animal.py
import random

#Yuck Look at this code!


#Animal class is the superclass, variables include the animal's name, if it's been fed, and if it's awake
class Animal:
    def __init__(self, name="", isAwake=false, isFed=false):
        name = name
        isAwake = isAwake
        isFed = isFed

	def sleep():
		print("Sleeping")

	def random_number_generator():
		num = random.randrange(2)
		return num

    def makeNoise():
        print("")

#Feline is a subclass of Animal
class Feline(Animal):
    def __init__(self, name="", isAwake=false, isFed=false):
        Animal.__init__(self, name="", isAwake=false, isFed=false)

	def roam():
		print("Wandering")

#Cat is a subclass of Feline
class Cat (Feline):
	def __init__(self, name="", isAwake=false, isFed=false):
        Feline.__init__(self, name="", isAwake=false, isFed=false)

    #The cat's action is unpredictable, hence the random action generator for when it makes a noise
	def random_action_generator():
		num = random_number_generator()
		if(num == 0)
			roam()
		else:
			print("Meow!")
	def makeNoise():
		random_action_generator()

#Lion is a subclass of Feline
class Lion(Feline):
	def __init__(self, name="", isAwake=false, isFed=false):
        Feline.__init__(self, name="", isAwake=false, isFed=false)
	def makeNoise():
		print("Roar")

#Tiger is a subclass of Feline
class Tiger (Feline)
	def __init__(self, name="", isAwake=false, isFed=false):
        Feline.__init__(self, name="", isAwake=false, isFed=false)
	def makeNoise():
		print("Grrrrrr")

#Canine is a subclass of Animal
class Canine (Animal):
    def __init__(self, name="", isAwake=false, isFed=false):
        Animal.__init__(self, name="", isAwake=false, isFed=false)
	def roam():
		print("Scampering")


#Wolf is a subclass of Canine
class Wolf (Canine):
	def __init__(self, name="", isAwake=false, isFed=false):
        Canine.__init__(self, name="", isAwake=false, isFed=false)
	def makeNoise():
		System.out.println("Grrrrr")


#Dog is a subclass of Canine
class Dog(Canine):
	def __init__(self, name="", isAwake=false, isFed=false):
        Canine.__init__(self, name="", isAwake=false, isFed=false)
	def makeNoise():
		print("Bark Bark")


#Pachyderm is a subclass of Animal
class Pachyderm(Animal):
    def __init__(self, name="", isAwake=false, isFed=false):
        Animal.__init__(self, name="", isAwake=false, isFed=false)
	def roam():
		print("Stompping")


#Hippo is a subclass of Pachyderm
class Hippo(Pachyderm):
	def __init__(self, name="", isAwake=false, isFed=false):
        Pachyderm.__init__(self, name="", isAwake=false, isFed=false)
	def makeNoise():
		print("Whaaaa")


#Rhino is a subclass of Pachyderm
class Rhino(Pachyderm):
    def __init__(self, name="", isAwake=false, isFed=false):
        Pachyderm.__init__(self, name="", isAwake=false, isFed=false)
    def makeNoise():
		print("Grumble Grumble")


#Elephant is a subclass of Pachyderm
class Elephant(Pachyderm):
    def __init__(self, name="", isAwake=false, isFed=false):
        Animal.__init__(self, name="", isAwake=false, isFed=false)
	def makeNoise():
		print("Raaaaaaa")
